eng/common/scripts/Rest-Methods.ps1

```diff
@@ -0,0 +1,124 @@
+. "${PSScriptRoot}\logging.ps1"
+
+function Invoke-RestMethodPost {
+  param (
+    [Parameter(Mandatory = $true)]
+    $apiURI,
+    [Parameter(Mandatory = $true)]
+    $body,
+    [Parameter(Mandatory = $true)]
+    $token
+  )
+
+  try {
+    if ($body.Count -gt 0) {
+      $resp = Invoke-RestMethod `
+      -Method POST `
+      -Body ($body | ConvertTo-Json) `
+      -Uri $apiURI `
+      -Headers (Get-RestMethodHeaders -token $token) `
+      -MaximumRetryCount 3
+  
+      return $resp
+    }
+    else {
+      $warning = "{0} with Uri [ $apiURI ] did not fire request because of empty body." -f (Get-PSCallStack)[1].FunctionName
```

Comment:
If the body is empty you should still support making a POST (it is technically valid).

Reply:
Yes I've not been a fan of this warning either I think folks should not call the method if their isn't a body.

Reply:
There cases where we want to to be a no-op because the body parameters are not passed in every case where the script is called. I guess we can also use conditions but this seemed neater to just log a warning and keep going.